src/Tizen.NET.MaterialComponents/Components/MSheets.cs

```diff
@@ -0,0 +1,244 @@
+            }
+            _frontSheet = contents;
+            if (_frontSheet != null)
+            {
+                _frontSheet.SetAlignment(-1, -1);
+                _frontSheet.SetWeight(1, 1);
+                if (!_frontSheet.IsVisible)
+                {
+                    _frontSheet.Show();
+                }
+                Panel.SetContent(_frontSheet, true);
+
+                UpdateScrim();
+                UpdateFrontSheetGeometry();
+            }
+        }
+
+        void UpdateFrontSheetGeometry()
+        {
+            if (_frontSheet != null)
+            {
+                _scrim.Geometry = Geometry;
+                Panel.Geometry = Geometry;
+            }
+        }
+
+        void Initialize(EvasObject parent, MSheetsDirection direction)
+        {
+            _backContainer = new Box(parent)
+            {
+                AlignmentX = -1,
+                AlignmentY = -1,
+                WeightX = 1,
+                WeightY = 1,
+            };
+            _backContainer.Show();
+            PackEnd(_backContainer);
+
+            _scrim = new Box(parent)
+            {
+                AlignmentX = -1,
+                AlignmentY = -1,
+                WeightX = 1,
+                WeightY = 1,
+                BackgroundColor = MColors.Current.OnSurfaceColor.WithAlpha(0.32)
+            };
+            _scrim.Show();
+            PackEnd(_scrim);
+
+            Panel = new Panel(parent);
+            Panel.IsOpen = false;
+            Panel.SetScrollable(true);
+            UpdateDirection(direction);
+            Panel.Toggled += (s, e) =>
+            {
+                UpdateScrim();
+                Toggled?.Invoke(this, e);
+            };
+            Panel.Show();
+            PackEnd(Panel);
+        }
+
+        void UpdateRatio(double ratio)
+        {
+            if (Panel.Direction == PanelDirection.Bottom)
+            {
+                if (ratio < _bottomSheetMinimumRatio)
+                {
+                    _ratio = _bottomSheetMinimumRatio;
+                }
+                else if (ratio > _bottomSheetDefaultRatio)
+                {
+                    _ratio = _bottomSheetDefaultRatio;
+                }
+                else
+                {
+                    _ratio = ratio;
+                }
+            }
+            else
+            {
+                if (ratio < _sideSheetMinimumRatio)
+                {
+                    _ratio = _sideSheetMinimumRatio;
+                }
+                else if (ratio > _sideSheetDefaultRatio)
+                {
+                    _ratio = _sideSheetDefaultRatio;
+                }
+                else
+                {
+                    _ratio = ratio;
+                }
+            }
+            Panel.SetScrollableArea(_ratio);
+        }
+
+        void UpdateScrim()
+        {
+            if (IsOpen)
+            {
+                _scrim.Show();
+                ((Window)this.Parent).Modal = true;
```

Comment:
how do you guarantee the type of parent is `Window`?

Reply:
removed